src/sentry/grouping/ingest/seer.py

```diff
@@ -241,16 +243,49 @@ def maybe_check_seer_for_matching_grouphash(event: Event) -> GroupHash | None:
             sample_rate=options.get("seer.similarity.metrics_sample_rate"),
             tags={"call_made": True, "blocker": "none"},
         )
+
         try:
             # If no matching group is found in Seer, we'll still get back result
             # metadata, but `seer_matched_grouphash` will be None
             seer_response_data, seer_matched_grouphash = get_seer_similar_issues(event)
-            event.data["seer_similarity"] = seer_response_data
-
-        # Insurance - in theory we shouldn't ever land here
-        except Exception as e:
+        except Exception as e:  # Insurance - in theory we shouldn't ever land here
             sentry_sdk.capture_exception(
                 e, tags={"event": event.event_id, "project": event.project.id}
             )
+            return None
+
+        # Find the GroupHash corresponding to the hash value sent to Seer
+        #
+        # TODO There shouldn't actually be more than one hash here, but
+        #   a) there's a bug in our precedence logic which leads both in-app and system stacktrace
+        #      hashes being marked as contributing and making it through to this point, and
+        #   b) because of how we used to compute secondary and primary hashes, we keep secondary
+        #      hashes even when we don't need them.
+        # Once those two problems are fixed, there will only be one hash passed to this function
+        # and we won't have to do this search to find the right one to update.
+        primary_hash = event.get_primary_hash()
+        grouphash_sent = list(
+            filter(lambda grouphash: grouphash.hash == primary_hash, all_grouphashes)
+        )[0]
+
+        # Update the relevant GroupHash with Seer results
+        metadata = grouphash_sent.metadata
```

Comment:
You mentioned that we're creating this elsewhere, but if we haven't created the row, is it worth using `update_or_create` here?

Reply:
No, because for now metadata creation is feature-gated and killswitched (and only happens for new `GroupHash` records. Only creating metadata records in one place lets us rely on that logic.

https://github.com/getsentry/sentry/blob/600065da5fe8ff2f1f2a677a3e1a76811a550527/src/sentry/grouping/ingest/hashing.py#L249-L255